engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
@sanjaypujare Please see JIRA. Currently, it has incorrect stack trace provided. The actual NPE that JIRA was filed for has stack trace similar to the issue that was fixed by f2bdef41fb35cc29e9abbf224f5ae49bf1b43b43.